packages/bpk-component-dialog/src/BpkDialog.tsx

```diff
@@ -53,31 +55,44 @@ const BpkDialog = ({
     );
   }
 
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape' && dismissible && onClose) {
+        onClose();
+      }
+    };
+
+    if (isOpen) {
+      window.addEventListener('keydown', handleKeyDown);
+    }
+
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [isOpen, onClose, dismissible]);
+
   return (
-    <Portal
```

Comment:
Have we validated that indeed there is no use case/ we do not need to render the dialog in a different part of the DOM tree?
For example, one use case for which we needed portals in the past was to place an element outside the main `pagewrap` element which all microsites render and to which our scrim component applies `aria-hidden` property so it results in the component being hidden from AT.

Reply:
We can setup a test or storybook example to test this! This works similar to the new modal v2, but you might be right now there is no `renderTarget` but I think the `withScrim` `getApplicationElement` property might be the main handler here, but you are right will test this!

Reply:
I did a check here and the property that does handle this with the `pagewrap` is the `getApplicationElement` property as I suspected so this will still work for consumers as this is what controlled where it goes and where the `aria-hidden` is applied to

Reply:
To confirm the line foudn it here whilst testing an example :) https://github.com/Skyscanner/backpack/pull/3508/files#diff-268f629c295e29c7e8875462a2fd7ef18a97eb0058f784296bddae08e1375635R74

Reply:
Yes, it's the scrim that applies the `aria-hidden` to the `getApplicationElement`, but it was the portal that was rendering it outside it.
Because looking at the [example](https://github.com/Skyscanner/backpack/pull/3508/files#diff-268f629c295e29c7e8875462a2fd7ef18a97eb0058f784296bddae08e1375635R63-R67), I can see that we explicitly render the dialog outside the `pagewrap` element so the examples would look good, but microsites would normally render it within the `pagewrap` element which sits at the very top for example here in [profile](https://github.skyscannertools.net/identity/profile/blob/e503cea49ba9a9c2b9209317936ffb6319464d15/app/routes/priceAlerts/index.html#L30)
Let me know if you'd like to zoom to chat through this 😄

Reply:
Taking a look at the code linked and live running of the website it seems that what we have in the examples and what we render is the exact same, whereby the dialog is rendered outside even when the pagewrap is the top element :) 

![Screenshot 2024-06-27 at 17 02 53](https://github.com/Skyscanner/backpack/assets/8831547/f9f4cfbf-f7c0-4b48-97c3-e04a181b5b43)